android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -67,28 +72,80 @@
   private static final String PWSTRIPTIME_KEY = "pwstriptime";
   private static final RegionResolver REGION_RESOLVER = new RegionResolver();
   private static final String SEPARATOR = "\0";
+  private static Set<String> mFavoriteUrls = new HashSet<>();
 
-  public static Comparator<PwPair> newDistanceComparator() {
-    return new Comparator<PwPair>() {
-      public Map<String, Double> cachedDistances = new HashMap<>();
+  // Compares PwPairs by first considering if it has been favorited
+  // and then considering distance
+  public static class PwPairComparator implements Comparator<PwPair> {
```

Comment:
I agree with Conley, it would be best to rename the comparator to something more descriptive.